src/parser/priest/discipline/modules/features/PurgeTheWicked.js

```diff
@@ -25,12 +28,41 @@ class PurgeTheWicked extends Analyzer {
     } else {
       this.dotSpell = SPELLS.SHADOW_WORD_PAIN;
     }
+
+    this.addEventListener(Events.cast.by(SELECTED_PLAYER).spell([SPELLS.PURGE_THE_WICKED_TALENT]), this.onDotCast);
+    this.addEventListener(Events.applydebuff.by(SELECTED_PLAYER).spell(SPELLS.PURGE_THE_WICKED_BUFF), this.onDotApply);
+    this.addEventListener(Events.refreshdebuff.by(SELECTED_PLAYER).spell(SPELLS.PURGE_THE_WICKED_BUFF), this.onDotApply);
   }
 
   get uptime() {
     return this.enemies.getBuffUptime(this.dotSpell.id) / this.owner.fightDuration;
   }
 
+  get extraPTWs() {
+    return this.ptwApplications - this.ptwCasts;
+  }
+
+  get ptwDamage() {
+    return this.abilityTracker.getAbility(SPELLS.PURGE_THE_WICKED_BUFF.id).damageEffective;
```

Comment:
I think you're not accounting for dying ennemies wich happens pretty often for PtW spreads. You're implying that all PtW spreads will always have their full duration wich is often not the case when it spreads to adds that are cleaved down pretty fast (Radiance, Orgozoa, Zacool). I think it would be safer to collect debuff applications that are after a penance cast and sum that damage of the events to that target before debuff fades.